go/interpreter/lfvm/gas.go

```diff
@@ -410,7 +401,17 @@ func gasSStoreEIP2929(c *context) (tosca.Gas, error) {
 		// If the caller cannot afford the cost, this change will be rolled back
 		c.context.AccessStorage(c.params.Recipient, slot)
 	}
-	value := tosca.Word(y.Bytes32())
+
+	////////////////////////////////////////////////////////////////////////
+	// If we fail the minimum gas availability invariant, fail (0)
+	if c.gas <= SstoreSentryGasEIP2200 {
+		c.status = statusOutOfGas
+		return 0, errNotEnoughGasReentrancy
+	}
+
+	// Gas sentry honoured, do the actual gas calculation based on the stored value
+	zero := tosca.Word{}
+	value := tosca.Word(c.stack.peekN(1).Bytes32())
```

Comment:
I would prefer that all peek operations are done at the same location. 
```go
x := tosca.Key(c.stack.peek().Bytes32())
y := tosca.Key(c.stack.peek().Bytes32())
```

Reply:
changes in this function reverted, since they had no effect and were part of a failed experiment.